api/src/opentrons/calibration_storage/file_operators.py

```diff
@@ -52,7 +72,7 @@ def read_cal_file(
 
 def save_to_file(
     filepath: StrPath,
-    data: typing.Mapping[str, typing.Any],
+    data: typing.Union[BaseModel, typing.Dict[str, typing.Any], typing.Any],
```

Comment:
That last `typing.Any` effectively removes any type constraints on this argument. Why is it needed?

```suggestion
    data: typing.Union[BaseModel, typing.Dict[str, typing.Any]],
```